src/adapters/cluster/redis/index.ts

```diff
@@ -14,14 +14,25 @@ class RedisClusterAdapter extends ClusterAdapter {
   }
 
   async connect(): Promise<this> {
-    return this._connect()
+    try{
+      return this._connect()
+    } catch (error) { 
+      console.error('Error connecting to Redis:', error)
+      throw error
+    }
   }
 
   async send(message: GleeMessage): Promise<void> {
+    try{
     return this._send(message)
+    } catch (error) {
+      console.error('Error sending message to Redis:', error)
+      throw error
+    }
   }
 
   async _connect(): Promise<this> {
+    try {
```

Comment:
we don't want to catch the error since we left the error handling the `_connect` method. errors thrown by this function will be handled by the `connect method.`
```suggestion
```